crates/red_knot_python_semantic/src/types.rs

```diff
@@ -2325,73 +2325,173 @@ impl<'db> Type<'db> {
     /// for y in x:
     ///     pass
     /// ```
-    fn try_iterate(self, db: &'db dyn Db) -> Result<Type<'db>, IterateError<'db>> {
+    fn try_iterate(self, db: &'db dyn Db) -> Result<Type<'db>, IterationError<'db>> {
```

Comment:
I'd prefer to keep the error type named `IterateError` because I then don't have to guess what the error type is named: `try_iterate` -> `IterateError`, `try_call` -> `CallError`.

Reply:
blegh, I much prefer `IterationError` 😄 because "iterate" just can't be used as a [noun adjunct](https://en.wikipedia.org/wiki/Noun_adjunct) in the same way as "call" or "bool" (because it's a verb rather than a noun -- "call" can be both a verb or a noun depending on context, which is why that feels OK to me, but that's not true for "iterate")

I don't feel too strongly though. I can change this back if you strongly dislike the new name, or if others agree with you :-)

Reply:
It's too much a detail to feel strongly about it but I'm interested in other opinions as well

Reply:
`IterateError` sounds wrong to me also, for the grammatical reason @AlexWaygood describes. I don't think I would be likely to make the assumption that the error type would be named precisely the same as the method that can return it. But I also don't feel strongly about it either way, `IterateError` really wouldn't bother me.